tests/e2e/base.py

```diff
@@ -37,6 +46,51 @@
 ]
 
 
+class Parameters:
+    param_d: Dict[str, Any]
+
+    def __init__(self, params: List[ParameterSet]) -> "Parameters":
+        self.param_d = {p.values[0]: p for p in params}
+
+    def _modify(self, *args: Union[str, str], func: Callable[[str], ParameterSet]) -> "Parameters":
+        for arg in args:
+            self.param_d[arg] = func(arg)
+        return self
+
+    def skip(self, *args: str, reason: Optional[str] = None) -> "Parameters":
+        return self._modify(*args, func=lambda x: pytest.param(x, marks=pytest.mark.skip(reason=reason)))
+
+    def xfail(
+        self, *args: str, reason: Optional[str] = None, strict=True, raises: Optional[Exception] = None
+    ) -> "Parameters":
+        return self._modify(
+            *args, func=lambda x: pytest.param(x, marks=pytest.mark.xfail(reason=reason, strict=strict, raises=raises))
+        )
+
+    def run(self, *args: str) -> "Parameters":
+        return self._modify(*args, func=lambda x: pytest.param(x))
+
+    def eval(self) -> List[ParameterSet]:
+        return list(self.param_d.values())
+
+
+class ParametersFactory:
```

Comment:
I'm wondering if it makes sense to have it as a class vs. to just have to functions - we don't have any state here, right? Just using it as a namespace for the methods